datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -71,18 +63,20 @@ impl Sum {
     }
 }
 
-/// Creates a [`PrimitiveGroupsAccumulator`] with the specified
-/// [`ArrowPrimitiveType`] which applies `$FN` to each element
-///
-/// [`ArrowPrimitiveType`]: arrow::datatypes::ArrowPrimitiveType
-macro_rules! instantiate_primitive_accumulator {
-    ($SELF:expr, $PRIMTYPE:ident, $FN:expr) => {{
-        Ok(Box::new(PrimitiveGroupsAccumulator::<$PRIMTYPE, _>::new(
-            &$SELF.data_type,
-            $FN,
-        )))
-    }};
+/// Sum only supports a subset of numeric types, instead relying on type coercion
```

Comment:
It might help to document what this macro does (aka calls helper macro given a s (what is s? The aggregate?))